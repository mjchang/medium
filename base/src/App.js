import React from 'react';

const App = () => {
  return <div>Dmo</div>;
};

export default App;
