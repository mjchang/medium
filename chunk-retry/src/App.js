import React, { lazy, Suspense } from 'react';

import importRetry from './importRetry';

import ErrorBoundary from './ErrorBoundary';

const Chunk = lazy(() =>
  importRetry(() => import(/* webpackChunkName: 'chunk' */ './Chunk'))
);

const App = () => {
  return (
    <ErrorBoundary>
      <Suspense fallback="Loading">
        <h1>Chunk Retry Demo</h1>
        <Chunk />
      </Suspense>
    </ErrorBoundary>
  );
};

export default App;
