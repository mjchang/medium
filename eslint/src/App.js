import React from 'react';
import PropTypes from 'prop-types';

/* eslint-disable */
import ReactDom from 'react-dom';
/* eslint-enable */

/* eslint-disable-line arrow-body-style */ const App = () => {
  return <h1>ESLint Demo</h1>;
};
App.propTypes = {
  /* eslint-disable-next-line react/no-unused-prop-types, react-redux/no-unused-prop-types */
  unused: PropTypes.string.isRequired,
};

export default App;
