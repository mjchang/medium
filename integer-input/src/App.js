import React from 'react';
import { useState } from 'react';

import IntegerInput from './IntegerInput';

const App = () => {
  const [value, setValue] = useState('');
  return (
    <>
      <h1>Integer Input</h1>
      <IntegerInput onChange={setValue} value={value} />
    </>
  );
};

export default App;
