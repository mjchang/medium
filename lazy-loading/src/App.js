import React, { lazy, Suspense } from 'react';

const PrefetchLoad = lazy(() =>
  import(
    /* webpackPrefetch: true, webpackChunkName: 'prefetchload' */ './PrefetchLoad'
  )
);

const LazyLoad = lazy(() =>
  import(/* webpackChunkName: 'lazyload' */ './LazyLoad')
);

const condition = false;

const App = () => {
  return (
    <Suspense fallback="Loading">
      <h1>Lazy Load Demo</h1>
      <LazyLoad />
      {condition && <PrefetchLoad />}
    </Suspense>
  );
};

export default App;
