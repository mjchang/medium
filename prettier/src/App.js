/* eslint-disable no-unused-vars */
import React from 'react';

const App = () => {
  /* prettier-ignore */
  const badFormat1 = {
    firstName: 'John', lastName: 'Doe', username: 'johndoe', email: 'johndoe@gmail.com', phone: '+1 (234) 567-8900', age: 32, sex: 'male',
  };

  /* prettier-ignore */
  const badFormat2 = {
    firstName: 'John',
    lastName: 'Doe', username: 'johndoe', email: 'johndoe@gmail.com', phone: '+1 (234) 567-8900', age: 32, sex: 'male',
  };

  const goodFormat = {
    firstName: 'John',
    lastName: 'Doe',
    username: 'johndoe',
    email: 'johndoe@gmail.com',
    phone: '+1 (234) 567-8900',
    age: 32,
    sex: 'male',
  };
  return <h1>Prettier Demo</h1>;
};

export default App;
