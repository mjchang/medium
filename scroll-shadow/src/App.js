import React from 'react';

import HorizontalCSSScrollShadow from './HorizontalCSSScrollShadow';
import HorizontalJSScrollShadow from './HorizontalJSScrollShadow';
import VerticalCSSScrollShadow from './VerticalCSSScrollShadow';
import VerticalJSScrollShadow from './VerticalJSScrollShadow';

const App = () => {
  return (
    <>
      <h1>Vertical CSS Scroll Shadow</h1>
      <VerticalCSSScrollShadow />
      <h1>Horizontal CSS Scroll Shadow</h1>
      <HorizontalCSSScrollShadow />
      <h1>Vertical JS Scroll Shadow</h1>
      <VerticalJSScrollShadow />
      <h1>Horizontal JS Scroll Shadow</h1>
      <HorizontalJSScrollShadow />
    </>
  );
};

export default App;
