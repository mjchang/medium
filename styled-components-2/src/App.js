import React from 'react';
import PropTypes from 'prop-types';

// use macro to achieve the same as babel-plugin-styled-components plugin
// must use this with create-react-app. If ejected, use babel-plugin-styled-components plugin
import styled, { createGlobalStyle } from 'styled-components/macro';

export const Button = styled.button`
  display: inline-block;
  padding: 6px 12px;
  font-size: 16px;
  font-family: Arial, sans-serif;
  line-height: 1.5;
  color: white;
  background-color: #6c757d;
  border: none;
  border-radius: 4px;

  :not(:disabled) {
    cursor: pointer;
  }

  :hover {
    background-color: #5a6268;
  }
`;

const StylePropButton = styled(Button)`
  ${props => props.$style ?? {}}
`;

// const App = () => {
//   return (
//     <StylePropButton
//       $style={{ backgroundColor: '#007bff' }}
//       onClick={() => alert('clicked!')}
//       type="button"
//     >
//       Button
//     </StylePropButton>
//   );
// };

// const App = () => {
//   return (
//     <Button as="a" href="https://styled-components.com/">
//       Link
//     </Button>
//   );
// };

const GlobalStyle = createGlobalStyle`
  *, *::before, *::after {
    box-sizing: border-box;
  }

  a {
    text-decoration: none;
  }
`;

// const App = () => {
//   return (
//     <>
//       <GlobalStyle />
//       <Button onClick={() => alert('clicked!')} type="button">
//         Link
//       </Button>
//     </>
//   );
// };

const AttributeButton = styled(Button).attrs(props => {
  const { type = 'button' } = props;
  return { type };
})``;

// const App = () => {
//   return (
//     <AttributeButton onClick={() => alert('clicked!')}>
//       Button
//     </AttributeButton>
//   );
// };

const DefaultPropsButton = styled(Button)``;
DefaultPropsButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
};
DefaultPropsButton.defaultProps = {
  type: 'button',
};

// const App = () => {
//   return (
//     <DefaultPropsButton onClick={() => alert('clicked!')}>
//       Button
//     </DefaultPropsButton>
//   );
// };

const App = () => {
  return (
    <div>
      <GlobalStyle />
      <h1>Styled Components Advanced Demo</h1>
      <div>
        <h2>Style Prop</h2>
        <StylePropButton
          $style={{ backgroundColor: '#28a745' }}
          onClick={() => alert('clicked!')}
          type="button"
        >
          Style Object
        </StylePropButton>
      </div>
      <div>
        <h2>Dynamic Components</h2>
        <Button as="a" href="https://styled-components.com/">
          Link
        </Button>
      </div>
      <div>
        <h2>Attributes</h2>
        <AttributeButton onClick={() => alert('clicked!')}>
          Attribute
        </AttributeButton>
      </div>
      <div>
        <h2>Default</h2>
        <DefaultPropsButton onClick={() => alert('clicked!')} type="button">
          Default Props
        </DefaultPropsButton>
      </div>
    </div>
  );
};

export default App;
