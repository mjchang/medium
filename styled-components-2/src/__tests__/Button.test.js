import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Button } from '../App';

test('snapshot', () => {
  const { container } = render(<Button />);
  expect(container.firstChild).toMatchSnapshot();
});

test('should have style', () => {
  const { container } = render(<Button />);
  expect(container.firstChild).toHaveStyleRule('background-color', '#6c757d');
  expect(container.firstChild).toHaveStyleRule('background-color', '#5a6268', {
    modifier: ':hover',
  });
});
