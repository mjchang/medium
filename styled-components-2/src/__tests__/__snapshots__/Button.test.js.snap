// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  display: inline-block;
  padding: 6px 12px;
  font-size: 16px;
  font-family: Arial,sans-serif;
  line-height: 1.5;
  color: white;
  background-color: #6c757d;
  border: none;
  border-radius: 4px;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:hover {
  background-color: #5a6268;
}

<button
  class="c0"
/>
`;
